scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1342,10 +1344,8 @@ private void sendMessage(final RawData message, final DTLSSession session) {
 					session.getSequenceNumber(),
 					new ApplicationMessage(message.getBytes(), message.getInetSocketAddress()),
 					session);
-			if (message.getMessageCallback() != null) {
-				message.getMessageCallback().onContextEstablished(ctx);
-			}
```

Comment:
I will change the order. But even with that change, everything must then be done synchronous to the callback. I hope this doesn't slow down too much.
(My idea was to catch IOExceptions or other issue and report them as onSendError instead of onSent.)